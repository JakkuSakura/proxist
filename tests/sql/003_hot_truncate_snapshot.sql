-- Append a few more ticks through proxist to exercise the hot seam.
INSERT INTO ticks (tenant, shard_id, symbol, ts, seq) VALUES
('alpha', 'alpha-shard', 'AAPL', toDateTime64('2024-01-01 10:00:02', 6), 3),
('alpha', 'alpha-shard', 'AAPL', toDateTime64('2024-01-01 10:00:03', 6), 4),
('alpha', 'alpha-shard', 'MSFT', toDateTime64('2024-01-01 10:00:04', 6), 5),
('beta', 'beta-shard', 'GOOG', toDateTime64('2024-01-01 10:00:05', 6), 2);

-- Show the merged ticks view proxist serves back.
SELECT tenant, symbol, count() AS total_rows, min(ts_micros) AS first_ts_micros, max(ts_micros) AS last_ts_micros
FROM ticks
GROUP BY tenant, symbol
ORDER BY tenant, symbol
FORMAT TSVWithNames;
