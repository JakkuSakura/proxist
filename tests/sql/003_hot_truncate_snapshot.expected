tenant	symbol	total_rows	first_ts_micros	last_ts_micros
alpha	AAPL	4	1704103200000000	1704103203000000
alpha	MSFT	2	1704103201000000	1704103204000000
beta	GOOG	2	1704103203000000	1704103205000000
